web/marketing/pages/free-offers.html

```diff
@@ -13,9 +13,10 @@
         <div class="col col-heading">Status</div>
     </div><hr>
     <div class="row offer-heading ">
-        <p class="offer-type">default offer</p>
+        <p class="offer-type">Default&nbsp;Offer</p>
     </div>
-    {{$defaultOffer := .FreeCredits.GetDefaultFromSet}}
+    {{$defaultOffer := .FreeCredits.Default}}
+    {{if $defaultOffer.Name}}
```

Comment:
If you use a pointer instead of struct then you can write it as:

```
{{ with $defaultOffer := .FreeCredits.Default }}
...
{{ end }}
```

Also the if should start a div above.

Reply:
We need the `<div>` section containing the `offer-heading` class  to render whether that section is empty or not because that's the header row. It renders properly that way too.

Reply:
I didn't like the way I checked if `Offer` was empty in the template so I created a new method to check it when template rendering.